### ISSUE TITLE
Quick actions add POI/note are off-center

### ISSUE THREAD

**Issue by matthijskooijman** [osmandapp/Osmand#9561]
When zooming (and a few seconds after zooming), a small circle is shown that (I presume) is the center of the map view and the zoom center point. When adding e.g. a note or POI using quick actions, a marker is shown where the note or POI will be added, bu this marker is not at the same point as the zoom/map center point.

![Screenshot_20200803-142141_OsmAnd~](https://user-images.githubusercontent.com/194491/89181911-e14cbc00-d594-11ea-9148-ab8ebcac5765.jpg)
![Screenshot_20200803-142145_OsmAnd~](https://user-images.githubusercontent.com/194491/89181913-e27de900-d594-11ea-8821-c86ffba8703e.jpg)

This is somewhat annoying, since in my workflow, I usually zoom towards the point I need, panning a bit between zoom actions to end up in the place I want to add the POI. The circle that is shown after zooming helps to pick the right point. Then, when opening the quick actions, the marker is shown a little further down, so I often have to move the marker a bit more after opening the quick actions, which is inconvenient. For this workflow, it would be better if the marker was shown at the (original) view center, rather than moving it down a bit.

I suspect the reason for this is that the marker is shown at the center of the map area that is not covered by the quick actions menu and is probably intentional. When the quick actions are open, the zoom center also moves to the same point. In a workflow where you open the quick actions menu first, and then zoom to the point you want to add it, it probably makes sense to put the marker and zoom point off-center (i.e. in the center of the visible map area), since that maximizes the amount of map shown around the marker.

I could do all the zooming and panning with the quick actions menu open, but then the screen is smaller, so it is a bit harder to navigate. Also, when I do this on my bike, there's a fair chance of accidentally tapping the screen, closing the quick actions menu again.

Would there be some way to support both workflows conveniently? One solution that might work is to actually move the map slightly when opening and closing the quick actions menu, so the (zoom) center and marker end up at the same point on the map before and after opening and closing the quick actions menu? The extra jump in map position might be unexpected, but since the center of the visible part of the map view would then stay the same, I would think this would be a nice approach?

I'm running OsmAnd version 3.7.4, on Android 9 on Galaxy Tab A (8.0", 2019), with offline maps and live updates.

**Comment by vshcherb** [comment]
Yes main point here is not to move the map, you can hit quick action and do zoom in/out after that, so center point will be aligned with marker

**Comment by matthijskooijman** [comment]
> Yes main point here is not to move the map, you can hit quick action and do zoom in/out after that, so center point will be aligned with marker

Yeah, I know I can move after hitting quick action, but as I said, that's not ideal:

> I could do all the zooming and panning with the quick actions menu open, but then the screen is smaller, so it is a bit harder to navigate. Also, when I do this on my bike, there's a fair chance of accidentally tapping the screen, closing the quick actions menu again.

(also, in practice I seem to simply forget this, so intuitively I find the spot first before opening the quick actions)

**Comment by matthijskooijman** [comment]
One more observation I forget: When you press the crosshair icon to move the map center to your current position, and *then* tap "quick actions", the map *is* actually moved (and back again) exactly as I proposed. So I guess there is already some code for that and I'm just suggesting to do this movement unconditionally rather than only when the map is centered on the current position?
